pandas/tools/plotting.py

```diff
@@ -3175,15 +3173,15 @@ def _grouped_plot_by_column(plotf, data, columns=None, by=None,
 
     _axes = _flatten(axes)
 
-    result = compat.OrderedDict()
+    result = Series()
     for i, col in enumerate(columns):
         ax = _axes[i]
         gp_col = grouped[col]
         keys, values = zip(*gp_col)
         re_plotf = plotf(keys, values, ax, **kwargs)
         ax.set_title(col)
         ax.set_xlabel(pprint_thing(by))
-        result[col] = re_plotf
+        result.loc[col] = re_plotf
```

Comment:
this is completely non-performant (copies each time). should append to lists (data/index) and create at the end (or create the Series at the end from the OrderedDict might be better/easier)